baseclasses/chemical_energy/uvvismeasurementconcentration.py

```diff
@@ -0,0 +1,218 @@
+        type=np.dtype(np.float64),
+        unit=('ug/ml'),
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='ug/ml'))
+
+    peak_value = Quantity(
+        type=np.dtype(np.float64),
+        description='The peak value of the wavelength-intensity graph. '
+                    'It can be set or it gets automatically computed when saving. '
+                    'To recompute it, remove the current value.',
+        a_eln=dict(component='NumberEditQuantity'))
+
+    peak_wavelength = Quantity(
+        type=np.dtype(np.float64),
+        unit='nm',
+        default=0,
+        description='The wavelength value of the automatically computed peak.',
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='nm'))
+
+    chemical_composition_or_formulas = Quantity(
+        type=str,
+        description='A list of the elements involved',
+        a_eln=dict(component='StringEditQuantity', label='Material'))
+
+    estimated_peak_center = Quantity(
+        type=np.dtype(np.float64),
+        unit='nm',
+        description='The wavelength value where the peak is estimated. This value is usually based on the material.',
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='nm'))
+
+    peak_search_range = Quantity(
+        type=np.dtype(np.float64),
+        unit='nm',
+        description='This value sets the range where to search for the peak around the estimated peak center.',
+        a_eln=dict(component='NumberEditQuantity', defaultDisplayUnit='nm'))
+
+    reference = Quantity(
+        type=Reference(Analysis.m_def),
+        description=('References the calibration that was used to calculate the concentration.'),
+        a_eln=dict(component='ReferenceEditQuantity', label='Concentration Detection'))
+
+    def normalize(self, archive, logger):
+        super(UVvisDataConcentration, self).normalize(archive, logger)
+
+        if self.chemical_composition_or_formulas:
+            if not archive.results:
+                archive.results = Results()
+            # if not archive.results.material:
+            archive.results.material = Material()
+            try:
+                from nomad.atomutils import Formula
+                formula = Formula(self.chemical_composition_or_formulas)
+                formula.populate(section=archive.results.material)
+            except Exception as e:
+                logger.warn('Could not analyse material', exc_info=e)
+
+        if self.chemical_composition_or_formulas == 'NH3':
+            if not self.estimated_peak_center:
+                self.estimated_peak_center = 650
+
+        if not self.peak_search_range:
+            self.peak_search_range = 20
+
+        if self.intensity is not None and self.wavelength is not None:
+            if self.chemical_composition_or_formulas is not None and self.peak_value is None:
+                if self.estimated_peak_center is not None and self.peak_search_range is not None:
```

Comment:
Hey @carla-terboven i tested it and it works so far fine. Thanks for that. One final thing for me would be that:
1. if you change the peak_wavelength manually this is chosen
2. if you change the estimated range or wavelength the whole calculation is redone

Reply:
I see how to resolve 2 by removing `and self.peak_value is None`.

For 1 I am a little clueless. Because if I add something like 
```
if self.peak_wavelength:
    self.estimated_peak_center = self.peak_wavelength
    self.peak_search_range = 0
```
this would cause a weird behavior for the following workflow:
1) setting the estimated peak center and range or use standard values of nh3 -> getting a peak
2) be unsatisfied with the peak and setting a different range or peak center -> nothing would happen because the first step already set a peak_wavelength which would be used directly with the code above. So when adapting the estimated peak center or the range the user would have to always delete the peak_wavelength.

Reply:
i would do it like follows: 
1. check if peak_wavelength is not set. if it is not set calcualte it as before if it is set do nothing.
2. take the peak_wavelength (calculated or inputted) and find the related intensity and set the intensity
3.  always redo the calculation for the concentration if a material is present

does this make sense?

Reply:
It makes sense but as soon as calculating the peak_wavelength once it is always set and the functionality with the peak center and range would not work any more. Only after deleting the old peak_wavelength there would be the option to use the range again. And I find it unintuitive for a user that the range and center is ignored all the time when not deleting the peak_wavelength. But if thats ok I can also change it to the workflow you suggested.

Reply:
your are right sry i missed that case! this is important it should then take the new range!
th eonly workaround would be to have another qunatity and hide it in the gui :D

Reply:
Oh wow this is not nice :D But it seems to be the only option indeed.

Reply:
yeah i also thought about it and actually this should be code which is in the GUI and not in the backend, then th is could be done with onChange callbacks in the browser

but i think this is out of our reach

Reply:
I agree.

I think then I would leave it as it is. If someone wants to set the peak_wavelength manually, they could use the estimated center and set the range to 0. This is not as nice as setting the peak_wavelength directly, but as a workaround I think it's still okay for a user.

Reply:
sure 
i am ok with that